src/interface.rs

```diff
@@ -10,6 +10,42 @@ use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 /// representing the IP for a Network Interface netmask
 pub type Netmask<T> = Option<T>;
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum Status {
+    Unknown,      // can not get status
+    Up,
+    Down,
+    Unavailable,  // not up or down
```

Comment:
Can we have these comments as as DocStrings?

```suggestion
    /// Could not retrieve status
    Unknown,
    Up,
    Down,
    // Status is not `up` nor `down`
    Unavailable,
```